File: 214. Shortest Palindrome/C++/Solution.cpp
class Solution {
public:
    string shortestPalindrome(string s) {
        string rev(s);
        reverse(rev.rbegin(), rev.rend());
        int length = s.length();
        for (int i = length; i >= 0; i--) {
            if (s.substr(0, i) == rev.substr(length - i)) {
                return rev.substr(0, length - i) + s;
            }
        }
        return "";
    }
};

File: 214. Shortest Palindrome/Java/Solution.java
class Solution {
    public String shortestPalindrome(String s) {
        String reverse = new StringBuilder(s).reverse().toString();
        for (int i = 0; i <= s.length(); i++) {
            if (s.startsWith(reverse.substring(i))) {
                return reverse.substring(0, i) + s;
            }
        }
        return null;
    }
}